indexing-service/src/main/java/io/druid/indexing/overlord/ThreadPoolTaskRunner.java

```diff
@@ -70,21 +72,34 @@ public ThreadPoolTaskRunner(
   )
   {
     this.toolboxFactory = Preconditions.checkNotNull(toolboxFactory, "toolboxFactory");
-    this.exec = MoreExecutors.listeningDecorator(Execs.singleThreaded("task-runner-%d"));
     this.conglomerate = conglomerate;
+    for (Task.Priority priority : Task.Priority.values()) {
+      this.exec.put(
```

Comment:
this changes the assumptions that all tasks are run in a single thread if multiple priorities are set. Is this going to be an issue? Is everything relying on that executor thread-safe?